imap/jmap_contact.c

```diff
@@ -6117,8 +6117,16 @@ static char *_value_to_uri_blobid(vcardproperty *prop,
                                   vcardproperty_version version,
                                   char **type, char **blobid)
 {
+    const vcardvalue *val;
+    const char *uri;
+
+    val = vcardproperty_get_value(prop);
+    if (vcardvalue_isa(val) == VCARD_X_VALUE)
+        uri = vcardvalue_get_x(val);
+    else
+        uri = vcardvalue_get_uri(val);
```

Comment:
This solves VALUE=X, but you could have VALUE=date:20151008 and then we'll crash because uri is null and callers expect it not to be.

Makes me thing this should never return null and xstrdupnull below is a mistake?